### PR TITLE
Improve ruby client with auto-reconnect, snake_case methods, and list…

### PR DESCRIPTION
**Pull request by peterl-stripe**
…ed methods for irb auto-complete

@ogdet could you review this (in particular for ruby naturalness on the client)?
CC @ppd-stripe @mark-stripe (I tried CCing @stripe-internal/teapot, but it didn't work, and another group creation attempt failed. A public teapot group is still a TODO for me).

### CODE REVIEW

File: ruby-package/lib/base_thrift_client.rb [hunk 1]
```diff
@@ -12,6 +12,7 @@ def initialize(server, port)
         @server = server
         @port = port
         @executor = init_executor(init_protocol())
+        @max_retries = 3
```
Comment (atausz-stripe) [inline ruby-package/lib/base_thrift_client.rb]: Maybe we can make this an optional parameter?

File: ruby-package/lib/base_thrift_client.rb [hunk 2]
```diff
@@ -32,6 +33,20 @@ def get_executor
...
+
+      def with_retries(&f)
+        (0..@max_retries).each do |retry_count|
+          begin
+            return f.call(get_executor)
+          rescue Thrift::TransportException, IOError
+            if retry_count == @max_retries then
+              raise # re-throw exception back to user
+            else
+              puts "Attempting to reconnect to #{@server}:#{@port}..."
```
Comment (atausz-stripe) [inline ruby-package/lib/base_thrift_client.rb]: Maybe we can also include the caller method name in the log message? http://stackoverflow.com/questions/5100299/how-to-get-the-name-of-the-calling-method

**Review by atausz-stripe — APPROVED** [review]

File: ruby-package/lib/base_thrift_client.rb (hunk 1 above)
Comment (peterl-stripe) [reply ruby-package/lib/base_thrift_client.rb]: Good idea.  I felt like I shouldn't hard-code a constant like this, but a full config file felt too heavyweight. I think optional parameter is best.

File: ruby-package/lib/base_thrift_client.rb (hunk 2 above)
Comment (peterl-stripe) [reply ruby-package/lib/base_thrift_client.rb]: Sounds good.  Once we're using it in ~production we should think about what logging would be most useful.

**Pull request merged**